app/uk/gov/hmrc/gform/gform/FormService.scala

```diff
@@ -62,4 +61,18 @@ class FormService {
       validatedType
     )
   }
+
+  private def toUpperCase(formValidatedData: List[FormComponentValidation]): List[FormComponentValidation] =
+    formValidatedData.map(isTextUpperCase)
+
+  def isTextUpperCase(formComponentValidation: FormComponentValidation): FormComponentValidation =
+    formComponentValidation.formComponent match {
+      case fc if fc.isUpperCase == false => formComponentValidation
+      case fc if fc.isUpperCase          => makeUpperCase(formComponentValidation)
+    }
+
+  private def makeUpperCase(formComponentValidation: FormComponentValidation) = formComponentValidation match {
+    case FormComponentValidation(formComponent, FieldOk(fieldValue, value)) =>
+      FormComponentValidation(formComponent, FieldOk(fieldValue, value.toUpperCase))
+  }
```

Comment:
This looks like possible `match may not be exhaustive` error.

We should handle also cases when `formFieldValidationResult` is something else than `FieldOk`.